src/components/token-rewards/ClaimAllRewardsDialog.tsx

```diff
@@ -34,24 +35,40 @@ export const ClaimAllRewardsDialog: FC<ClaimAllRewardsDialogProps> = ({
   const { mode } = useDarkMode()
   const { isMobile } = useBreakPoint()
   const { claimableRewardsWithTokens, refreshRewards } = useBoostedRewards()
-  const { sendTransaction, createTransactionBuilder } = useTransaction()
+  const { createTransactionBuilder, sendBatchTransaction } = useTransaction()
   const { connection } = useConnectionConfig()
   const { GammaProgram } = usePriceFeedFarm()
   const { publicKey: userPublicKey } = useWalletBalance()
+  const {wallet} = useWallet();
 
   const claimAllMutation = useMutation({
     mutationFn: async () => {
-      const txBuilder = createTransactionBuilder()
+      // Split rewards into batches of 5
+      const batchSize = 5
+      const rewards = claimableRewardsWithTokens.rewards
+      const batches = []
 
-      for (const reward of claimableRewardsWithTokens.rewards) {
-        for (const rewardInfo of reward.rewards) {
-          const tx = await claimRewards(GammaProgram, userPublicKey, connection, rewardInfo)
-          txBuilder.add(tx)
+      for (let i = 0; i < rewards.length; i += batchSize) {
+        batches.push(rewards.slice(i, i + batchSize))
+      }
+
+      const transactions = [];
+
+      // Process each batch
+      for (const batch of batches) {
+        const batchTxBuilder = createTransactionBuilder()
+
+        for (const reward of batch) {
+          console.log(reward);
+          const tx = await claimRewards(GammaProgram, userPublicKey, connection, reward)
```

Comment:
claimRewards function needs `BoostedRewardInfo` type where `reward` is being passed.